tools/src/launcher/Main.hs

```diff
@@ -515,11 +508,49 @@ spawnNode (path, args, mbLogPath) = do
             logInfo "Node has started"
             return (ph, asc)
 
-runWallet :: (FilePath, [Text]) -> M ExitCode
-runWallet (path, args) = do
+runWallet :: Bool -> (FilePath, [Text], Maybe FilePath) -> M ExitCode
+runWallet shouldLog (path, args, mLogPath) = do
     logNotice "Starting the wallet"
-    view _1 <$>
-        liftIO (readProcessWithExitCode path (map toString args) mempty)
+    phvar <- newEmptyMVar
+    liftIO $ case mLogPath of
+        Just lp -> do
+            cr <- createLogFileProc path args lp
+            system' phvar cr mempty EWallet
+        Nothing ->
+           let cr = if shouldLog then
+                        Process.CreatePipe
+                    else Process.NoStream
+           in system' phvar (createProc cr path args) mempty EWallet
+
+createLogFileProc :: FilePath -> [Text] -> FilePath -> IO Process.CreateProcess
+createLogFileProc path args lp = do
+    _ <- createDirectoryIfMissing True (directory lp)
+    (_, logHandle) <- (lp,) <$> openFile lp AppendMode
+    liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
+    return $ createProc (Process.UseHandle logHandle) path args
+
+createProc :: Process.StdStream -> FilePath -> [Text] -> Process.CreateProcess
+createProc stdStream path args =
+    (Process.proc path (map toString args))
+            { Process.std_in = Process.CreatePipe
+            , Process.std_out = stdStream
+            , Process.std_err = stdStream
+            }
+
+asyncLog :: Handle -> Handle -> ProcessHandle -> Executable -> IO ExitCode
+asyncLog stdO stdE pH nt = do
+    withAsync (forever $ T.hGetLine stdO >>= customLogger nt) $ \_ ->
+        withAsync (forever $ T.hGetLine stdE >>= customLogger nt) $ \_ -> do
+        waitForProcess pH
+
+customLogger :: Executable -> Text -> IO ()
+customLogger loggerName logStr = do
+    putStrLn $ logNameStr <> logStr
```

Comment:
As a small nitpick: you can pass some value denoting whether `logStr` is from `stdout` or `stderr` and print to `stdout` or `stderr` here. Currently all output, even `stderr`, will be printed to `stdout`.